### ISSUE TITLE
[Feature Request]: UI Enhancement – Reduce Unused Space, Improve Accessibility, and Clarify Metrics

### ISSUE THREAD

**Issue by mxair95000** [meshtastic/Meshtastic-Android#2933]
### Contact Details

.mxair95000 (discord)

## **Brief**

**I’ve identified some UX problems and drafted potential solutions, with a prototype and design file attached below. These proposals are preliminary and intended to spark discussion. I’d love to hear feedback from both users and developers 😺**

Prototype Link 👉👉👉 ：https://www.figma.com/proto/Wrz2kY3xc5nzwo3yXksEYQ/Meshtastic-UI?page-id=0%3A1&node-id=43-770&p=f&viewport=-2703%2C862%2C0.23&t=n34gQF27SkA3qvEI-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=43%3A770

## Problem 01: Excessive Unused Space on Node Page
On the current Node Page, around **18%–29% of the layout space is unused and preversd for node infomation display.**  As a result, it squeezes the usable area and the displayed text becomes too small and harder to read. Also, there are a lot of infomation in limited space at same time, making the node page overwhelming.

<img width="960" height="633" alt="Image" src="https://github.com/user-attachments/assets/7d257ab5-8944-434b-aa96-f51d12a67e8e" />

## → Proposed Enhancement 1-1 : Use "swipeable info bar" to display metrics
As shown in the attached mockup, I suggest displaying a node’s information in a horizontal, swipeable bar, where all metrics are aligned inline. This allows users to check each metric easily by swiping. If any metric is unavailable, the following metric automatically shifts forward, making space usage more efficient without compromising text size or readability.

![Image](https://github.com/user-attachments/assets/8fb14305-6cf7-44a8-a87f-c02a39476bae)

<img width="6714" height="1935" alt="Image" src="https://github.com/user-attachments/assets/92d9e752-80cb-4d08-81c4-880a8f6d758f" />

----

## **Problem 02: Low Text Contrast from Dynamic Card Colors**
In the 2.6 update, Node Page card colors now change based on the Short ID Name Badge. I assume this was designed to help users distinguish between nodes. However, as shown in the figure below, it often reduces text contrast, making the content harder to read, especially for the components that use StatusColor（semantic color） to express meaning, like Lora signal metric and coordinate link.

<img width="2346" height="1380" alt="Image" src="https://github.com/user-attachments/assets/c42f5cb2-2d96-407a-8d0b-7703122610c0" />

### **→ Proposed Enhancement 2-1 : Apply Fixed Light & Dark Backgrounds for Cards**
To improve consistency, I recommend switching the background color scheme to only Light and Dark modes, as in the previous app version. This prevents the contrast ratio from fluctuating when card colors change, ensuring stable readability. At the same time, I’ve increased the size and weight of the short ID badges, so users can still distinguish between nodes at a glance even without varied background colors.

- **Light Theme:** surfaceContainerLowestLight = Color(0xFFFFFFFF)
- **Dark Theme:** surfaceContainerDark = Color(0xFF1C211C)

<img width="1156" height="486" alt="Image" src="https://github.com/user-attachments/assets/67ae9fad-2832-45ba-9894-4729e0ca882b" />

### → Proposed Enhancement 2-2 : Assign Dark Mode Color for Hyperlink
Currently, hyperlinks don’t have separate colors for Light and Dark modes, which results in insufficient contrast on light backgrounds (e.g., coordinates). As shown in the figure, I suggest adding a dedicated dark-mode variant of *HyperlinkBlue*. This would significantly improve contrast in Light mode and ensure compliance with WCAG AA standards.

- **Light Theme:** HyperlinkBlue = Color(0xFF43C3B0)
- **（New）Dark Theme:** HyperlinkBlue = Color(0xFF1F6157)

<img width="2610" height="1044" alt="Image" src="https://github.com/user-attachments/assets/fe861e54-ef02-4da0-bfcf-cfcbe2553810" />

### **→ Proposed Enhancement 2-3 : Add Colored Containers Behind Signal Indicators to improve contrast**
I suggest modifying the StatusColor scheme and adding a background container to ensure that the LoRa signal indicators meet the required contrast ratio under WCAG 2.1 AA, as shown in the figure below.The colors in my design come from the [Material Design Baseline palette](https://m3.material.io/styles/color/static/baseline#b5a485b5-ee5f-4890-b7a2-ead284121e37) and can be applied directly via tokens.

<img width="1491" height="585" alt="Image" src="https://github.com/user-attachments/assets/dec289f0-f658-4b73-8755-8aa0f0b51bff" />

<img width="2610" height="3252" alt="Image" src="https://github.com/user-attachments/assets/368b2448-7940-456f-8036-7ebb6c9bfd99" />

----

## **Problem 03: Confusing Metrics for Beginners**
At present, the Node Card shows over 15 metrics, but several of them have no titles because of space constraints. In addition, metrics like *ChUtil / AirUtilTx, SNNR, and SNR* are not self-explanatory, making it challenging for new users to understand what the numbers actually mean without checking the documentation.

### **→ Proposed Enhancement 3-1 : Add Icons and short title**
With the additional space made available in **Proposed Enhancement 1-1 (Swipeable Infobar)**, each metric could be paired with a short title and an icon. This would make the information easier to recognize at a glance. All icons are based on the Material Icons set.

### **→ Proposed Enhancement 3-2 : Add Tooltips and Explanations**
For metrics that may be harder to interpret, such as *ChUtil*, *AirUtilTx*, or *Satellite*. I suggest adding tooltips with short, in-context explanations. Users can click (or hover) on a metric to see a brief description. I’ve drafted some sample content, and the exact wording can be further refined through discussion.

![Image](https://github.com/user-attachments/assets/5c40dabd-3f90-486f-b828-1cf0d01dc4c1)

----

## **Problem 04: Unclear Entry Points to Detail Page**
Currently, the **ID badge** functions as the button to open the Node Detail Page. However, using the badge as a button is unintuitive, since there is no clear visual cue that it is clickable.

<img width="1515" height="708" alt="Image" src="https://github.com/user-attachments/assets/dfbfe00f-0a93-4a77-b603-f531568d987c" />

### **→ Proposed Enhancement 4-1 : Add an Arrow Button as a Visual Cue**
I suggest adding a small **arrow button** at the left side of the card, similar to the iOS design. This provides a clear visual affordance, making it obvious to users that tapping the card leads to the detail page.

<img width="1315" height="1080" alt="Image" src="https://github.com/user-attachments/assets/8a2b6450-2bc6-47b7-8e6d-ff9de5d65909" />

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by jamesarich** [comment]
Some decent ideas in here, but also some Android antipatterns - the ` >` carat being one of those.

Have you validated your changes against Material 3 Expressive and Android Design Guidelines? We try to use base components and best practices for Android (and Android users) as we can, and if we're looking to make changes we should be following those guidelines.

**Comment by pedjas** [comment]
Unused space is even greater when phone is oriented horizontally. I guess in that case screen could be divided in two so one half shows list of nodes and other shows map or messages.

**Comment by weirdal3333** [comment]
That qrcode button covers up the information in the last node on the list, in both horizontal and vertical orientations. A little bit of a buffer at the bottom of the list, or an option to hide the qr code would be a good addition to this issue.

**Comment by jamesarich** [comment]
> That qrcode button covers up the information in the last node on the list, in both horizontal and vertical orientations. A little bit of a buffer at the bottom of the list, or an option to hide the qr code would be a good addition to this issue.

https://github.com/meshtastic/Meshtastic-Android/pull/2949
